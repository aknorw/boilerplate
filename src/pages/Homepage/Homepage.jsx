import React from 'react'
import PropTypes from 'prop-types'
import { Helmet } from 'react-helmet'
import { connect } from 'react-redux'

import { setFoo } from 'services/basic/actions'

// This container does not follow the rule about DOM markup
// This is intentional as this is only temporary

import Button from 'components/Button'

export const Homepage = ({ handleButtonClick }) => (
  <div>
    <Helmet title="Homepage" />
    <h1>
      Homepage
    </h1>
    <Button onClick={handleButtonClick}>
      Change foo from Homepage
    </Button>
  </div>
)

Homepage.propTypes = {
  handleButtonClick: PropTypes.func.isRequired,
}

const mapDispatchToProps = dispatch => ({
  handleButtonClick: () => dispatch(setFoo()),
})

export default connect(null, mapDispatchToProps)(Homepage)
