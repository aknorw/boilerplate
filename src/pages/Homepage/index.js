import ConnectedHomepage from './Homepage'

export default ConnectedHomepage
