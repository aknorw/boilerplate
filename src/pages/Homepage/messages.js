import { defineMessages } from 'react-intl'

export default defineMessages({
  title: {
    id: 'pages.HomePage.title',
    defaultMessage: 'Welcome',
  },
  subtitle: {
    id: 'pages.HomePage.subtitle',
    defaultMessage: 'Some interesting things to be written here',
  },
  city: {
    id: 'pages.HomePage.city',
    defaultMessage: 'City',
  },
  temperature: {
    id: 'pages.HomePage.temperature',
    defaultMessage: 'Temperature',
  },
})
