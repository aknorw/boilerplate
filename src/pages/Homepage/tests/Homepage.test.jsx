import React from 'react'
import { shallow } from 'enzyme'

import { Homepage } from '../Homepage'

const mockFn = jest.fn()
console.error = jest.fn()

describe('<Homepage />', () => {
  it('should be defined', () => {
    expect(Homepage).toBeDefined()
  })
  it('should log 1 error when no props are passed', () => {
    shallow(<Homepage />)
    expect(console.error).toHaveBeenCalledTimes(1)
  })
  it('should render correctly', () => {
    const tree = shallow(<Homepage handleButtonClick={mockFn} />)
    expect(tree).toMatchSnapshot()
  })
})
