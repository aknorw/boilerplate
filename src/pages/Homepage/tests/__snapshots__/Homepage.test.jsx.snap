// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> should render correctly 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Homepage"
  />
  <h1>
    Homepage
  </h1>
  <Button
    onClick={[MockFunction]}
  >
    Change foo from Homepage
  </Button>
</div>
`;
